roles/self_hosted_first_host/tasks/main.yml

```diff
@@ -47,77 +49,83 @@
       group: qemu
-  tags:
-    - setup
 
-- name: Get the answer file over there
-  template:
-      src: answers.j2
-      dest: '{{ config_dir }}/answers'
-      mode: 0600
   tags:
     - setup
 
-- name: Create cloud init temp directory
-  file:
-      path: '{{ config_dir }}/cloud_init'
-      state: directory
-      mode: 0700
-      owner: qemu
-      group: qemu
+- name: Find the path to the appliance image
+  shell: find /usr/share/ovirt-engine-appliance -name '*-appliance-*.ova'
+  register: engine_appliance
   tags:
     - setup
 
-- name: Copy over the cloud init data
+- name: Get the answer file over there
   template:
-      src: '{{ item.src }}'
-      dest: '{{ item.dest }}'
-      owner: qemu
-      group: qemu
+      src: answers.j2
+      dest: '{{ config_dir }}/answers'
       mode: 0600
-  with_items:
-    - { src: user-data.j2, dest: '{{ config_dir }}/cloud_init/user-data' }
-    - { src: meta-data.j2, dest: '{{ config_dir }}/cloud_init/meta-data' }
   tags:
     - setup
 
-- name: Generate cloud-init iso
-  shell: |
-    genisoimage -output '{{ config_dir }}/cloud_init/cloud.iso' -volid cidata -joliet -rock -input-charset utf-8  '{{ config_dir }}/cloud_init/meta-data' '{{ config_dir }}/cloud_init/user-data'
-  tags:
-    - setup
-
-- name: Fix permissions on iso
-  file:
-      name: '{{ config_dir }}/cloud_init/cloud.iso'
-      group: qemu
-      owner: qemu
-      state: file
-      mode: 0600
-  tags:
-    - setup
+- block:
+  - block:
+    - name: Copy authorised ssh keys from the hypervisor
+      command: cat ~/.ssh/authorized_keys
+      register: hypervisor_authorized_keys
+
+    - name: Set authorized ssh key if not provided
+      set_fact: ssh_authorized_keys={{ hypervisor_authorized_keys.stdout_lines }}
+    when: ssh_authorized_keys == []
+
+
+  - name: Create cloud init temp directory
+    file:
+        path: '{{ config_dir }}/cloud_init'
+        state: directory
+        mode: 0700
+        owner: qemu
+        group: qemu
+    tags:
+      - setup
+
+  - name: Copy over the cloud init data
+    template:
+        src: '{{ cloud_init_file.src }}'
+        dest: '{{ cloud_init_file.dest }}'
+        owner: qemu
+        group: qemu
+        mode: 0600
+    with_items:
+      - { src: user-data.j2, dest: '{{ config_dir }}/cloud_init/user-data' }
+      - { src: meta-data.j2, dest: '{{ config_dir }}/cloud_init/meta-data' }
+    loop_control:
+      loop_var: cloud_init_file
+    tags:
+      - setup
+
+  - name: Generate cloud-init iso
+    shell: |
+      genisoimage -output '{{ config_dir }}/cloud_init/cloud.iso' -volid cidata -joliet -rock -input-charset utf-8  '{{ config_dir }}/cloud_init/meta-data' '{{ config_dir }}/cloud_init/user-data'
+    tags:
+      - setup
+
+  - name: Fix permissions on iso
+    file:
+        name: '{{ config_dir }}/cloud_init/cloud.iso'
+        group: qemu
+        owner: qemu
+        state: file
+        mode: 0600
+    tags:
+      - setup
+  when: not ansible_deploy
```

Comment:
minor stylistic nit, but for large blocks like this I prefer to put the when at the top for the sake of readability, like

```yaml
- when: not ansible_deploy
  block:
     - name: do_stuff
       ...
```

Reply:
Nice and easy change. 
but when I come to do that, it looks me better to leave it like this because of the "when" bound the block and IMHO it's looking more readable where is the end of the block special when its long block.
wdyt?

Reply:
I just have a hard time telling which block the when is applying to, and in the future it's easy to make some bugs by accidentally adding or removing a few spaces.

Reply:
Done!!